### ISSUE TITLE
No Frontline between Sir Abu and Al Dafra

### ISSUE THREAD

**Issue by Fleck85** [dcs-liberation/dcs_liberation#6]
Campaign: Persian Gulf full map
Sides: BLUEFOR Modern vs. North Korea 2000
Version: Development build of 16. Jun 2020
Save file: [liberation save files.zip](https://github.com/Khopa/dcs_liberation/files/4791713/liberation.save.files.zip)

Issue:
After Capturing the first Base (Al Dafra) there is no frontline between the island base on Sir Abu and Al Dafra.
This leads to an error creating the ground conflict. (Save Game attached)

Proposed Solution:
Quite hard to say, as it needs a design decision, because there is no sense creating a frontline between an offshore and onshore island base for a ground based attack.
It may be possible to generate naval units war (including buying ships at suitable bases) similar to the ground units. It would add a nice aspect to the gameplay and add more variety. Pathfinding of naval vessels should also be much easier than with ground units.

**Comment by Khopa** [comment]
Oh yes indeed, I hadn't tested the PG full map in a while. (This is a legacy map that i kept from the old version).

It probably does not work very well performance-wise with the newest version anyway ? I was actually thinking about removing the "Full Maps" entirely for performance reasons.

BTW, I'll start by removing the "frontlines" between land and islands that do not make sense.

> Quite hard to say, as it needs a design decision, because there is no sense creating a frontline between an offshore and onshore island base for a ground based attack.

Yes i agree, in the newest map i did not add frontlines when there is no land, as it does not make sense.

> It may be possible to generate naval units war (including buying ships at suitable bases) similar to the ground units. It would add a nice aspect to the gameplay and add more variety. Pathfinding of naval vessels should also be much easier than with ground units.

That could be a possibility, but i think it will end up being weird. In my experience in DCS, most ships tends to fire missiles at everything and kill each other at 50-100nm+ range, so I believe it would be really hard to balance properly.

I have thought about making the Carrier and LHA movable on the campaign map in the future. So you'd be able to recruit a limited amount of ground units to store on the LHA, and then you would move it on the campaign map so it is close enough to an enemy base. And when it is near the enemy base, the LHA would send its ground units to capture it. 

Basically what would happen in DCS, would be that i'd spawn a "fast assault boat" near the LHA for each group of vehicles. And these small ships would head toward the enemy base coast, there, there will a trigger zone, and if the ships reach this trigger zone, it will make the vehicles appear on the coast. And then these vehicles will proceed toward the enemy base in an attempt to capture it.

**Comment by Fleck85** [comment]
The performance on my side was not great, but it was borderline acceptable (your milage may vary).
I am flying in VR and I dont have a high end system, so I am not blaming the mission itself.
(I7 4790k, TitanX, 24GB DDR3 Ram, SSD)
Usually I get around 20 fps in busy missions, in some lighter missions 45 fps. 
The first missions in liberation dropped to 13 fps, which is barely playble, but it got better with the following missions.
Edit:
The performance settings in liberation were all enabled. (Smoke on, Moving units on...)

Performance wise I would rather suggest some kind of "pruning" like it is done in the DCE framework. Some very distant units could be removed for example in order to get better fps. Perhaps configurable via the performance settings...

On the naval war:
The engagement ranges and the constant firing of missiles until the countermeasures are depleted one side could really be a performance killer. I have not enough experience in DCS naval missions to really think about a suitable solution.

But the idea of moving your carrier from turn to turn is great! and being able to capture the islands via assoult boats is also very nice.

I wil try to help the debugging as far as i can with my limited knowledge. Especiallit the whole GIT thingy is quite new for me...

**Comment by MysticDaedra** [comment]
@Fleck85, you have a Titan X and 24GB of ram _and_ VR, but don't consider your system to be "high-end"? I hate to break it to you, but the second you slapped 24GB and a Titan X in there you graduated from "mid-range" lol. 

I was also thinking about having naval warfare possibilities. Even on the smaller PG maps, there are no connections where there definitely should be. UAE has two connecting airbases, but they don't connect to anything else. Additionally, there's an island base that could connect differently as well, especially with some form of naval front. Would grant us more Harpoon options as well ;)

**Comment by Khopa** [comment]
> you have a Titan X and 24GB of ram and VR, but don't consider your system to be "high-end"? 

Well, it's not high-end enough to play a busy DCS mission in 60 FPS, especially in VR :grin:

My config is : 
* GTX 1070 8GB
* 32GB RAM DDR4 
* i5-6600K oc @4.1Ghz
* Win 10 and DCS installed on SSD

I'm running fine in 1080p. The" North Caucasus DCS liberation campaign" can run at 40/45 FPS with the last version in single player, which is fine enough for me, and about the same performance that i get in MP servers.

***

> Performance wise I would rather suggest some kind of "pruning" like it is done in the DCE framework. Some very distant units could be removed for example in order to get better fps. Perhaps configurable via the performance settings...

Yes, i had that in mind as well, and I have implemented a basic logic to remove distant units from the generated mission :  

Considering this DCS Liberation campaign on Caucasus Full map. 
(Which is an awful scenario for performance, that i'm pretty sure would make DCS crash on my PC)

![image](https://user-images.githubusercontent.com/2546901/85016331-d1eeec00-b169-11ea-8b13-adb142155d05.png)
 
This is what will be generated with a culling distance set up to 100km (the minimum value) : 

![image](https://user-images.githubusercontent.com/2546901/85016030-3fe6e380-b169-11ea-8771-fb0fce6a7a73.png)

Without culling, this is what is generated : 

![image](https://user-images.githubusercontent.com/2546901/85016064-4d9c6900-b169-11ea-9038-b2fbbbae22ed.png)

This will definitely improve performance on large map, at the cost of a bit of realism. But hopefully this will make it playable for lower configurations, and will be configurable in settings :

![image](https://user-images.githubusercontent.com/2546901/85017264-7c1b4380-b16b-11ea-9d08-59f4c565f82b.png)

The reference point for the culling calcultation, is the current frontline(s). Anything more than 100km away from the frontline(s) will not be generated. If there is no ground units frontline, the reference point chosen is the middle point between the closest opposing airbases;

Of course it could be improved and it's quite dumb for now :
 
* Instead of a dumb algorithm using the distance to the conflict, i should maybe spawn the "x" closest group to the conflict instead.
* What if the player select a target that is off the generator range ? (Spoiler: the target will not have spawn, and you will not be able to destroy it)

**Comment by TezzaCourtney** [comment]
I find most of the performance hits and stutters when each side starts large scale shooting and flinging lots of missiles at each other.

Perhaps placing a limit on the number of flights that will be generated might help (Configurable by user)
And a limit on the number of ground units that are generated, or taking part in front line action. Of course some units can be sat in reserve behind the front line not doing much (BAI Target)

I like your thinking on the culling distance though, it is a good start :-)

**Comment by DanAlbert** [comment]
This campaign was redone, and no longer includes Sir Abu at all.
